controllers/datadogdashboard/dashboard.go

```diff
@@ -15,9 +16,25 @@ import (
 	"github.com/go-logr/logr"
 )
 
-// Dashboard
+// Transform v1alpha1 dashboard into a datadogV1 Dashboard
 func buildDashboard(logger logr.Logger, ddb *v1alpha1.DatadogDashboard) *datadogV1.Dashboard {
-	// create a dashboard
+	// If there is a JSON string, use that to build the dashboard instead
+	if ddb.Spec.Data != "" {
+		dashboard := &datadogV1.Dashboard{}
+		json.Unmarshal([]byte(ddb.Spec.Data), dashboard)
+		// Overrides. Title, NotifyList and TemplateVariables takes precedence over what's in the json.
+		if ddb.Spec.Title != "" {
```

Comment:
This can be refactored to avoid duplication of settings these fields. You create and init dashboard in one way or other, and then apply common settings.